cf-key/cf-key.c

```diff
@@ -354,6 +354,29 @@ static GenericAgentConfig *CheckOpts(int argc, char **argv)
             }
 
         case 'C':
+            /* Hack to get optional- and required arguments with short option
+             * to work the same way.
+             *
+             *     If the option has an optional argument, it must be written
+             *     directly after the option character if present.
+             *      - getopt(1) - Linux man page
+             *
+             * Confusing, right? This little hack makes sure that the option
+             * character and the optional argument can also be separated by a
+             * whitespace character. This way optinal- and required arguments
+             * can be passed in the same fashion.
+             *
+             * How it works:
+             * If optarg is NULL, we check whether the next string in argv is
+             * an option. If this is not the case, we assume it is an
+             * argument. The argument is extracted and optind is advanced
+             * accordingly.
+             */
+            if (optarg == NULL && argv[optind] != NULL
+                && argv[optind][0] != '-')
```

Comment:
So `optind` is the *next* item? I'd expect it to be the *current* item. Probably worth adding to the comment.